src/createServer.js

```diff
@@ -1,3 +1,70 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+
+const { detectCase } = require('./convertToCase/detectCase.js');
+const { convertToCase } = require('./convertToCase/convertToCase.js');
+
+const caseVariants = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    const normalizedUrl = new URL(req.url, `http://${req.headers.host}`);
+    const toCase = normalizedUrl.searchParams.get('toCase');
+    const textToConvert = normalizedUrl.pathname.slice(1);
+    const errors = [];
+
+    res.setHeader('Content-Type', 'application/json');
+
+    if (!textToConvert) {
+      errors.push({
+        message:
+          'Text to convert is required. Correct request is: '
+          + '"/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!toCase) {
+      errors.push({
+        message:
+        '"toCase" query param is required. Correct request is: '
+        + '"/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (caseVariants.indexOf(toCase) === -1 && toCase) {
+      errors.push({
+        message:
+          'This case is not supported. '
+          + 'Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+      });
+    }
+
+    if (errors.length) {
+      res.statusMessage = 'Bad request';
+      res.statusCode = '400';
+
+      res.end(JSON.stringify({ errors }));
+    }
+
+    if (!errors.length) {
+      const newConvertedText = convertToCase(textToConvert, toCase);
+
+      res.statusCode = '200';
+      res.statusMessage = 'OK';
+
+      const response = {
+        ...newConvertedText,
+        originalCase: detectCase(textToConvert),
```

Comment:
check what is returned from `convertToCase` function

Reply:
sorry, but I can't understand what do you mean here?